### DISCUSSION TITLE
How to add a new tabitem in txt2img_header?

### DISCUSSION THREAD

**Discussion by samuel25gh** [anapnoe/stable-diffusion-webui-ux#216] [Q&A] [answered]
![image](https://github.com/anapnoe/stable-diffusion-webui-ux/assets/90382562/0a3530db-28fb-40ed-8e7e-09bf084f167c)

Hi. I would like to add a custom tabitem in the txt2img_header for txt2img and make a custom sub-tab of my own (a new one along side from Prompts, Parameters, Scripts / Extensions). Where in the project files should I be looking into in order to do so? Thank you.

**Comment by anapnoe** [comment, marked as answer]
Hi you should look at the `template-txt2img-params.html` 
for the button add one and define the tabview-tabItemId and the tabGroup
tabItemId acts as a target container and tabGroup as a shorthand iterator to show the active one from that group

Create a button entry to the layout
```
<button title="Parameters" tabItemId="#txt2img_params_tabitem" tabGroup="txt2img" id="txt2img_params" class="xtabs-tab shrink">
<div class="mask-icon icon-equalize">
</div>
<span>Params</span>
</button>
```

Create a view to display your module 
```
 <div id="txt2img_tabitem_group" class="flexbox col">
    <div id="txt2img_params_tabitem" class="xtabs-item">
        <div data-selector="#txt2img_script_container" data-parent-selector="#tab_txt2img" class="portal no-bg">
        </div>
    </div>
</div>
```
